### ISSUE TITLE
storing credentials in env var

### ISSUE THREAD

**Issue by noahdietz** [30x/project-management#130]
We need a way for customers to store credentials securely in their namespace, and retrieve them (via something like a secret).

**Comment by noahdietz** [comment]
this was brought up by Martin while discussing build-time & deployment time environment variables

i.e. an app needing database credentials as environment variables

**Comment by whitlockjc** [comment]
Secrets are the way to do it.  But going down this route means providing k8s API access in an abstracted way, something we didn't want to do.  I agree with this for now but we really should plan on giving direct k8s access for post-beta for things where it makes sense, like secret management.  _(This is exactly the type of situation we've known would come and it's here sooner than we thought.)_

**Comment by whitlockjc** [comment]
I think that this highlights the problems we will quickly run into with the current Shipyard design:
1. To support this, Shipyard would have to abstract/proxy the Kubernetes Secrets API(s)
2. The current model of using a PTS URL does not allow for referencing Secrets

Maybe @mpnally can comment more on this.

**Comment by mpnally** [comment]
1. Is a dup of the Authz web hook (#138)
2. This probably means a kiln upgrade to store PTSpecs instead of faking them in URLs.

**Comment by noahdietz** [comment]
@mpnally how is this a dup of authz web hook? I thought the purpose of this issue was to address client-app specific credentials. Authz web hook is for k8s API access...

**Comment by whitlockjc** [comment]
I agree with @noahdietz, I think @mpnally is crossing the streams.  The purpose for this work is to give Shipyard users the ability to store secure information in Kubernetes and reference it as part of their Shipyard deployments.  The authz/webhook support is different.
